samples/capture-to-file/encode-worker.js

```diff
@@ -0,0 +1,86 @@
+importScripts('./webm-writer2.js')
+
+let webmWriter = null;
+let fileWritableStream = null;
+let frameReader = null;
+
+async function startRecording(fileHandle, frameStream, trackSettings) {
+  let frameCounter = 0;
+
+  fileWritableStream = await fileHandle.createWritable();
+
+  webmWriter = new WebMWriter({
+      fileWriter: fileWritableStream,
+      codec: 'VP9',
+      width: trackSettings.width,
+      height: trackSettings.height});
+
+  frameReader = frameStream.getReader();
+
+  const init = {
+    output: (chunk) => {
+      webmWriter.addFrame(chunk);
+    },
+    error: (e) => {
+      console.log(e.message);
+      vtr.stop();
+    }
+  };
+
+  const config = {
+    codec: "vp09.00.10.08",
+    width: trackSettings.width,
+    height: trackSettings.height,
+    bitrate: 10e6,
+  };
+
+  let encoder = new VideoEncoder(init);
+  let support = await VideoEncoder.isConfigSupported(config);
+  console.assert(support.supported);
+  encoder.configure(config);
+
+  frameReader.read().then(async function processFrame({done, value}) {
+    let frame = value;
+
+    if(done) {
+      await encoder.flush();
+      encoder.close();
```

Comment:
Does this also need a call to `frameReader.releaseLock()` or `frameReader.cancel()`?

Reply:
I think no. The trigger for us entering this branch should be that stopRecording() was called higher in the stack, which calls frameReader.cancel(), which sets done = true.

For this demo I think cancel() (vs releaseLock()) is the right call because we have no need to use the readableStream (cancel it) after recording stops. Starting a new recording makes a new worker w/ a new stream.